From 0d2d362cb195c86de3a901f4591858f31f1dd26a Mon Sep 17 00:00:00 2001
From: V1tr10l7 <vitriol1744@gmail.com>
Date: Tue, 18 Feb 2025 02:56:35 +0100
Subject: [PATCH] [cryptix]: add missing includes for termios headers

---
 cmatrix.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/cmatrix.c b/cmatrix.c
index 90284cb..5cac226 100644
--- a/cmatrix.c
+++ b/cmatrix.c
@@ -25,11 +25,13 @@
 #include <stdlib.h>
 #include <stdarg.h>
 #include <string.h>
+#include <strings.h>
 #include <time.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
-#include <termios.h>
+#include <asm/termbits.h>
+#include <asm/termios.h>
 #include <signal.h>
 #include <locale.h>
 
@@ -51,12 +53,6 @@
 #include <unistd.h>
 #endif
 
-#ifdef HAVE_TERMIOS_H
-#include <termios.h>
-#elif defined(HAVE_TERMIO_H)
-#include <termio.h>
-#endif
-
 /* Matrix typedef */
 typedef struct cmatrix {
     int val;
-- 
2.48.1

