From 9aa37e1a328b23a1bda744260ed4d52307b370f9 Mon Sep 17 00:00:00 2001
From: V1tr10l7 <vitriol1744@gmail.com>
Date: Wed, 17 Sep 2025 03:51:56 +0200
Subject: [PATCH] add cryptix target

---
 support/config.sub | 1 +
 1 file changed, 1 insertion(+)

diff --git a/support/config.sub b/support/config.sub
index 4aaae46..07abf13 100755
--- a/support/config.sub
+++ b/support/config.sub
@@ -1989,6 +1989,7 @@ case $os in
 	| cnk* \
 	| conix* \
 	| cos* \
+	| cryptix* \
 	| cxux* \
 	| cygwin* \
 	| darwin* \
-- 
2.51.0

