From 0715fd562ed903e05d578789bdec9622ae84e541 Mon Sep 17 00:00:00 2001
From: V1tr10l7 <vitriol1744@gmail.com>
Date: Mon, 24 Mar 2025 04:17:06 +0100
Subject: [PATCH] [cryptix]: add SYS_MUNMAP definition

---
 sysdeps/cryptix/include/cryptix/syscall.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sysdeps/cryptix/include/cryptix/syscall.h b/sysdeps/cryptix/include/cryptix/syscall.h
index d56f4c44..93c1b784 100644
--- a/sysdeps/cryptix/include/cryptix/syscall.h
+++ b/sysdeps/cryptix/include/cryptix/syscall.h
@@ -29,6 +29,7 @@ constexpr size_t SYS_FSTAT         = 5;
 constexpr size_t SYS_LSTAT         = 6;
 constexpr size_t SYS_LSEEK         = 8;
 constexpr size_t SYS_MMAP          = 9;
+constexpr size_t SYS_MUNMAP        = 11;
 constexpr size_t SYS_IOCTL         = 16;
 constexpr size_t SYS_ACCESS        = 21;
 constexpr size_t SYS_PIPE          = 22;
-- 
2.48.1

