diff --color -Naur ncurses-orig/config.sub ncurses-work/config.sub
--- ncurses-orig/config.sub	2022-08-01 10:05:29.000000000 +0200
+++ ncurses-work/config.sub	2025-01-21 20:29:08.773162798 +0100
@@ -1754,7 +1754,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | cryptix*)
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff --color -Naur ncurses-orig/configure ncurses-work/configure
--- ncurses-orig/configure	2022-11-05 22:13:26.000000000 +0100
+++ ncurses-work/configure	2025-01-21 20:30:47.580560252 +0100
@@ -6510,6 +6510,10 @@
 		LINK_PROGS="$SHELL ${rel_builddir}/mk_prog.sh"
 		LINK_TESTS="$SHELL ${rel_builddir}/mk_prog.sh"
 		;;
+    (cryptix*)
+        CC_SHARED_OPTS='-fPIC'
+                    MK_SHARED_LIB='${CC} -shared -o $@'
+        ;;
 	(mingw*)
 		cf_cv_shlib_version=mingw
 		cf_cv_shlib_version_infix=mingw
diff --color -Naur ncurses-orig/test/picsmap.c ncurses-work/test/picsmap.c
--- ncurses-orig/test/picsmap.c	2022-12-04 01:40:11.000000000 +0100
+++ ncurses-work/test/picsmap.c	2025-01-21 20:31:38.217161220 +0100
@@ -60,6 +60,8 @@
 #define my_intptr_t	long
 #endif
 
+#include <strings.h>
+
 #if HAVE_TSEARCH
 #include <search.h>
 #endif
